docker/docker_image_dest.go

```diff
@@ -38,12 +39,29 @@ func newImageDestination(ctx *types.SystemContext, ref dockerReference) (types.I
 	if err != nil {
 		return nil, err
 	}
+	if ctx.DockerMount != nil {
```

Comment:
A `*types.SystemContext` can always be `nil`, and should be treated as if it were `&types.SystemContext{}`.